Refactor/doc tests

neomonkeus opened the pull request:
@niftools/pyffi-reviewer 

# Overview
Convert remaining text based doctests to run using nose.

## Fixes Known Issues
N\A

## Documentation
Covered by existing documentation

## Testing
All existing test cases passing

### Manual
N\A pure automation

### Automated
Full regression suite passing

## Additional Information

neomonkeus committed:
Refactor optimise vertex cache to nose test.

Renamed files to drop the name asa already convention from dir structure

neomonkeus committed:
Refactor optimisation spell to remove zero scale

neomonkeus committed:
Refactor optimisation spell to remove unused bones

neomonkeus committed:
Remove unused directory

neomonkeus committed:
Move base file class to util module.

neomonkeus committed:
Migrate clamp material alpha to nose.

neomonkeus committed:
Migrate string palette cleanup spell to nose.

neomonkeus committed:
Migrate detach havok tristripsdata spell to nose.

neomonkeus committed:
Migrate detach havok tristripsdata spell to nose.

neomonkeus committed:
Apply some more formatting changes

neomonkeus committed:
Migrate tangent space fix spell tests to nose

neomonkeus committed:
Migrate texture path spell test to nose.

neomonkeus committed:
Migrate modify bone priorities path spell test to nose.

neomonkeus committed:
Fix bug where file was reading original file instead of copy.

neomonkeus committed:
Migrate delete branch spell test to nose.

neomonkeus committed:
Migrate delete vertex colors spell test to nose.

neomonkeus committed:
Migrate fix substitute string palette spell test to nose.

neomonkeus committed:
Migrate fix vertex weight influence test to nose.

neomonkeus committed:
Migrate fix vertex weight influence test to nose.

neomonkeus committed:
Migrate dump texture information spell test to nose.

neomonkeus committed:
Make the base file source generic to allow for format types.

neomonkeus committed:
Add a utility function to invoke cgftoaster

neomonkeus committed:
Execute basic cgftoaster spells

neomonkeus committed:
Move file reading to specific format class

neomonkeus committed:
Migrated cgf dump file test to nose

neomonkeus committed:
Convert vcol cgf tests to nose and delete doctest.txt file for cgf

neomonkeus committed:
Update setup.py for tests and remove pep line length

neomonkeus commented:
Travis automation has picked up issues with python 3.7 and some generator functions.
Need to investigate if there is a backwards compatible update that will support 3.4 - 3.7

TagnumElite commented:
I figured out the problem, my solution was [this](https://github.com/TagnumElite/pyffi/commit/0fb6fd3bbcf84bfc27592ce0af302b646050a3ec). Or surround the `next()` with `try-catch` `StopIteration` clause

Example:
```python
def get_triangles(self):
    """Generator for all triangles."""
        if self.faces: ...
        elif self.indices_data:
        it = iter(self.indices_data.indices)
        try:
            while True:
                yield next(it), next(it), next(it)
        except StopIteration:
            pass
```

neomonkeus committed:
Merge branch 'develop' into refactor/doc_tests

HENDRIX-ZT2 committed:
Fix: weights in partition instead of skindata

Fixes skinning for Wildlife Park 2&3 - would scale mesh to zero. Could wrap the actual skinning in a helper function to re-use code and save some lines.

HENDRIX-ZT2 committed:
Fix: Import bhkLimitedHingeConstraint

Had no attribute sub_constraint, causing import to fail due to apply_scale() being called on it.

HENDRIX-ZT2 committed:
Fix: Import bhkMalleableConstraint

Not tested because I didn't find one in the wild, but it has the exact same pattern found in bhkLimitedHingeConstraint, and no sub_constraint attribute is noted in the nif.xml, but the corrected ones are.

neomonkeus committed:
Merge pull request #2 from HENDRIX-ZT2/refactor/doc_tests

Essential PyFFI fixes for NIF import
